providers/src/airflow/providers/amazon/aws/operators/emr.py

```diff
@@ -666,6 +670,7 @@ def __init__(
         job_flow_overrides: str | dict[str, Any] | None = None,
         region_name: str | None = None,
         wait_for_completion: bool = False,
+        wait_for_steps_completion: bool = False,
```

Comment:
I think both `wait_for_completion` and `wait_for_steps_completion` can be very confusing.
If we are to support the new functionality I suggest to introduce a new parameter `wait_policy` which will accept value of `wait_for_steps_completion`, `wait_for_steps_completed_and_machine_terminated` similar to what we did with Kubernetes in https://github.com/apache/airflow/pull/30718 when we had more than just a bool option for delete pod.

Reply:
I like that idea!

Reply:
@eladkal I prefer your idea too. I've added a version of what you said, but it might not be what you had in mind.

When you say `wait_for_steps_completed_and_machine_terminated` what is the waiter name you think should map to it?